src/agents/base_agent.py

```
@@ -0,0 +1,161 @@
+from sklearn.model_selection import train_test_split
+from stable_baselines3.common.vec_env import DummyVecEnv
+from src.utils.environment import get_env
+from src.callbacks.log_test_callback import LogTestCallback, LOG_DIR
+from sklearn.model_selection import TimeSeriesSplit
+
+
+class BaseAgent:
+    def __init__(
+        self,
+        experiment_name: str = "Base Agent",
+        data: pd.DataFrame = [],
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)